### PR TITLE
resource and license apis

### LINKED ISSUE #512: api endpoint for creating resources
- [ ] api endpoint that takes as parameters at least id and resource licenses

### PR DESCRIPTION
**Pull request by opqdonut**
fixes #512

### CODE REVIEW

File: src/clj/rems/api/schema.clj [hunk 1]
```diff
@@ -16,9 +16,33 @@
...
+   :localizations {s/Keyword {:title s/Str :textcontent s/Str}}})
+
+(def ResourceLicense
+  {:id s/Num
+   :licensetype (s/enum "text" "link" "attachment")
+   :start DateTime
+   :end (s/maybe DateTime)
+   :title s/Str
+   :textcontent s/Str
+   :localizations {s/Keyword {:title s/Str :textcontent s/Str}}})
```
Comment (opqdonut) [inline src/clj/rems/api/schema.clj]: should we define this as

```
(def ResourceLicense
  (merge License {:start DateTime :end (s/maybe DateTime)})

File: src/clj/rems/api/schema.clj [hunk 2]
```diff
@@ -16,9 +16,33 @@
...
+
+(def Resource
+  {:id s/Num
+   :modifieruserid s/Str
+   :resid s/Str
+   :start DateTime
+   :end (s/maybe DateTime)
+   :licenses [ResourceLicense]})
+
+(def ApplicationLicense
```
Comment (opqdonut) [inline src/clj/rems/api/schema.clj]: this could also be a merge

File: resources/sql/queries.sql [hunk 3]
```diff
@@ -33,11 +33,22 @@ INSERT INTO catalogue_item
...
+  endt
+FROM resource
+
 -- :name create-resource! :insert
 -- :doc Create a single resource
 INSERT INTO resource
 (id, resid, prefix, modifieruserid)
-VALUES (:id, :resid, :prefix, :modifieruserid)
+VALUES (
+  /*~ (if (:id params) */ :id /*~ */ DEFAULT /*~ ) ~*/,
```
Comment (tjalkane) [inline resources/sql/queries.sql]: Should `id` parameter be removed from here?

File: resources/sql/queries.sql [hunk 4]
```diff
@@ -33,11 +33,22 @@ INSERT INTO catalogue_item
 (title, formid, resid, wfid)
 VALUES (:title, :form, :resid, :wfid)
 
+-- :name get-resources :? :*
+SELECT
+  id,
+  modifieruserid,
+  resid,
```
Comment (tjalkane) [inline resources/sql/queries.sql]: `prefix` field should be added to this.

File: src/clj/rems/api/schema.clj [hunk 5]
```diff
@@ -16,9 +16,33 @@
    :localizations (s/maybe {s/Any s/Any})})
 
 (def License
+  {:id s/Num
+   :licensetype (s/enum "text" "link" "attachment")
+   :title s/Str
+   :textcontent s/Str
+   :localizations {s/Keyword {:title s/Str :textcontent s/Str}}})
```
Comment (tjalkane) [inline src/clj/rems/api/schema.clj]: It's good to know that license also has start & end even though they are not used

File: resources/sql/queries.sql (hunk 4 above)
Comment (opqdonut) [reply resources/sql/queries.sql]: done

File: resources/sql/queries.sql (hunk 3 above)
Comment (opqdonut) [reply resources/sql/queries.sql]: done

File: src/clj/rems/api/schema.clj (hunk 5 above)
Comment (opqdonut) [reply src/clj/rems/api/schema.clj]: done

File: src/clj/rems/api/schema.clj (hunk 1 above)
Comment (opqdonut) [reply src/clj/rems/api/schema.clj]: done

File: src/clj/rems/api/schema.clj (hunk 2 above)
Comment (opqdonut) [reply src/clj/rems/api/schema.clj]: done

**Pull request merged**